### DISCUSSION TITLE
How to style marker color with css

### DISCUSSION THREAD

**Discussion by lovetox** [xyflow/xyflow#4908] [Q&A] [answered]
Hi,

I found the options to style the edge path via `stroke`. But this seems to not affect the marker. Any hints how to change the color with css? Thanks

**Comment by bcakmakoglu** [comment, marked as answer]
> Any hints how to change the color with css? Thanks

You can do something like this 
```css
.react-flow__marker marker polyline {
  fill: pink !important;
  stroke: pink !important;
}
```

[An example can be found here.](https://codesandbox.io/p/sandbox/sharp-frost-nzgdkd?file=%2Fsrc%2Fstyles.css%3A14%2C1-18%2C1)

But I'm assuming you'd rather want *dynamic* coloring for the edge marker.

To do that you can use an object to define the edge marker, for example:
```ts
  {
    id: "c->d",
    source: "c",
    target: "d",
    animated: true,
    markerEnd: {
      type: MarkerType.ArrowClosed,
	  color: 'pink',
    },
  },
```

Now you can update the color in the edge using, for example, [`updateEdge`](https://reactflow.dev/api-reference/hooks/use-react-flow#update-edge), together with the edges' stroke.

```ts
const { updateEdge } = useReactFlow()

updateEdge(edgeId, (prevEde) => {
	return {
		...prevEdge,
		style: {
			stroke: 'pink',
		},
		markerEnd: {
			...prevEdge.markerEnd,
			color: 'pink',
		}
	}
})
```

If you need *more* flexibility for this, you can use a custom edge marker *inside* of a Custom Edge component and style the marker directly yourself, for example if an edge is selected.
To re-use the existing edge markers you could just copy paste their source code.

[Here's an example.](https://codesandbox.io/p/sandbox/crazy-fermat-v3f4c9?workspaceId=ws_7iZPUfEyQC7Podbs7wwbxV)

**Reply by lovetox** [reply]
Thanks, my goal is to style the marker with a different color when the edge is selected, i guess i need to do the CustomEdge approach. I would be very interested in the example you linked but it yields a 404 for me.

**Reply by bcakmakoglu** [reply]
Try the sandbox again, I turned it public now :)
